scripts/scripts.js

```diff
@@ -72,11 +72,28 @@ async function loadEager(doc) {
   }
 }
 
+/**
+ * sets the Content-Security-Policy meta tag to the document based on JSON file
+ */
+
+async function setCSP() {
+  const resp = await fetch(`${window.hlx.codeBasePath}/scripts/csp.json`);
+  const json = await resp.json();
+  const directives = Object.keys(json);
+  const policy = directives.map((directive) => `${directive} ${json[directive].join(' ')}`).join('; ');
+  const meta = document.createElement('meta');
+  meta.setAttribute('http-equiv', 'Content-Security-Policy');
+  meta.setAttribute('content', policy);
+  document.head.appendChild(meta);
+}
+
 /**
  * Loads everything that doesn't need to be delayed.
  * @param {Element} doc The container element
  */
 async function loadLazy(doc) {
+  await setCSP();
```

Comment:
I'd check if this is effective in blocking eagerly loaded resources that violate the CSP.

Reply:
i am not sure that i am really that worried about the CSPs protecting block code...
to me this probably has to be in place before we execute untrusted code, eg. martech stack, so one could argue that this is in the wrong place altogether. so to a certain extent i think i am saying i don't even know when this should be loaded, so i am not really sure that an `await` really makes a difference here...

i am somewhat doubtful of the allowlist based approach altogether, so maybe a nonce based `strict-dynamic`  approach would be the right thing here, but i definitely don't know enough about the repercussions of that vis-a-vis the somewhat questionable benefit of CSPs around XSS attacks on franklin sites.

Reply:
https://research.google/pubs/pub45542/

Reply:
Revisiting this, I think this is right, but only the first step. The next step should be @chicharr's lightweight tag management controlling the CSP

Reply:
I think `setCSP` should be called as the first thing in `loadPage`, so the CSPs can offer protection as soon as possible. The additional request to fetch the configuration could be avoided in the critical path of FCP/LCP by inlining the configuration in the function. Something like:

```javascript
function setCSP() {
  const config = {
    "default-src": [
        "'self'"
    ],
    ...
  };

  const directives = Object.keys(config);
  const policy = directives.map((directive) => `${directive} ${config[directive].join(' ')}`).join('; ');
 ...
}
```